rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java

```diff
@@ -161,12 +161,13 @@ int inQueue(
             "local count = 0\n" +
             "for i, channel in ipairs(channels) do \n" +
             "  local readyQueue = channel .. \":ready_queue\"\n" +
-            "  count = count + redis.call('ZCARD', readyQueue)\n" +
+            "  count = count + redis.call('ZCOUNT', readyQueue, 0, $currentTimeMillis$)\n" +
```

Comment:
watch out with this - zcount isn't constant time. things could get dicey if you have a whole whole bunch of jobs in ready.

Reply:
true, but there's no other way to get a count. in omega we cache the result of this for a short time as well.